manifests/repo.pp

```diff
@@ -18,6 +18,7 @@
   String            $proxy         = 'absent',
   Integer           $version       = 7,
   Optional[String]  $base_repo_url = undef,
+  Optional[String]  $key_source    = 'https://artifacts.elastic.co/GPG-KEY-elasticsearch'
```

Comment:
CI failure can be fixed by removing the `Optional`.  While here, maybe we can use a stricter type than `String`, e.g. `String[1]` (non-empty string) or `Stdlib::HTTPUrl` (which would not allow `puppet:///modules/profile/esGPG-KEY-elasticsearch/` so maybe not so good?).  Also, missing coma at end of line :wink: 

```suggestion
  Stdlib::HTTPUrl   $key_source    = 'https://artifacts.elastic.co/GPG-KEY-elasticsearch',
```